tests/test_flows.py

```diff
@@ -2364,6 +2403,44 @@ def my_flow():
 
 
 class TestFlowHooksOnFailure:
+    def test_on_failure_hooks_is_iterable(self):
+        def failure_hook():
+            pass
+
+        with pytest.raises(
+            TypeError, match="Expected iterable for 'on_failure'; got function instead."
+        ):
+
+            @flow(on_failure=failure_hook)
+            def flow1():
+                pass
+
+        @flow(on_failure=[])
+        def flow2():
+            pass
+
+        assert flow2.on_failure == []
```

Comment:
I would expect this to raise. Also, it's probably best to move this to a separate test.

Reply:
I think default value for `on_failure` is `[ ]`, so isn't is allowed to have empty list, please help me understand.

Reply:
Got it, I assumed default value to be empty list which is not true it is `None`

Reply:
In `flow.py` and `tasks.py` we are not concerned about empty list as values, I think I need to update code there first.

Reply:
Let's check for the empty list first:

```python
if hooks is not None:
  if not hooks:  # This is equivalent to checking if len(hooks) == 0 for lists
      raise ValueError(f"Empty list passed for '{hook_name}'")
  try:
      hooks = list(hooks)
```

Reply:
Sure, it's done.